wamr-compiler/main.c

```diff
@@ -738,6 +765,7 @@ main(int argc, char *argv[])
     if (option.custom_sections) {
         free(option.custom_sections);
     }
+    free(llvm_options);
```

Comment:
Is it better to check llvm_options and free it only it is not NULL? like L765

Reply:
why?

Reply:
When `--mllvm=xxx` isn't specified, llvm_options is NULL, no need to free it in fact. Though the linux manual of free says no operation is performed if ptr is NULL, I am not sure whether it is a good coding style, we usually free a buffer only when it is NULL. It is good if the same behavior is ensured in other platforms.

Reply:
free(NULL) is guaranteed to be a no-op by C language.
i'm not aware of any problematic platforms wrt that.

Reply:
OK, thanks.